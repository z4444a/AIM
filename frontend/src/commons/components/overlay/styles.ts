import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';

export const styles = (theme: Theme) =>
  createStyles({
    container: {
      width: '100vw',
      height: '100vh',
      position: 'fixed',
      zIndex: 10000,
      backgroundColor: 'rgba(100,100,100,0.5)',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
    innerContainer: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
    },
  });

export type Style = WithStyles<'container' | 'innerContainer'>;
