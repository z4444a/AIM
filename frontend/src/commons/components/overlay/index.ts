import withStyles from '@material-ui/core/styles/withStyles';
import { Overlay } from './overlay';
import { styles } from './styles';

export default withStyles(styles)(Overlay);
