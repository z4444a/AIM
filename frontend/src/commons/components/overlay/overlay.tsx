import React, { ReactNode } from 'react';
import { Style } from './styles';
import Spinner from '../spinner/spinner';
import { Typography } from '@material-ui/core';

export interface Props {
  active?: boolean;
  text?: string;
}

export type InternalProps = Props & Style;

export class Overlay extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { classes, active, text } = this.props;
    if (active) {
      return (
        <div className={classes.container}>
          <div className={classes.innerContainer}>
            <Spinner />
            {text && <Typography>{text}</Typography>}
          </div>
        </div>
      );
    }
    return null;
  }
}
