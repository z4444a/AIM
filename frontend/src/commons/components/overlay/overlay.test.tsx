import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { InternalProps, Overlay } from './overlay';

describe('Overlay', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const props: InternalProps = {
      classes: {
        container: 'container',
        innerContainer: 'innerContainer',
      },
      active: true,
      text: 'Overlay works!!!',
    };
    wrapper = shallow(<Overlay {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
