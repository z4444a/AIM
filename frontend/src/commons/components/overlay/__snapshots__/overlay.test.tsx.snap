// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="innerContainer"
  >
    <Spinner />
    <WithStyles(Typography)>
      Overlay works!!!
    </WithStyles(Typography)>
  </div>
</div>
`;
